receiver/oracledbreceiver/scraper.go

```diff
@@ -0,0 +1,306 @@
+	sessionExecuteCountSQL        = "select ss.SID as session_id, se.value as VALUE  from v$session ss, v$sesstat se, v$statname sn where se.STATISTIC# = sn.STATISTIC# and se.SID = ss.SID and NAME = 'execute count'"
+	sessionParseCountTotalSQL     = "select ss.SID as session_id, se.value as VALUE from v$session ss, v$sesstat se, v$statname sn where se.STATISTIC# = sn.STATISTIC# and se.SID = ss.SID and NAME = 'parse count (total)'"
+	sessionUserCommitsSQL         = "select ss.SID as session_id, se.value as VALUE from v$session ss, v$sesstat se, v$statname sn where se.STATISTIC# = sn.STATISTIC# and se.SID = ss.SID and NAME = 'user commits'"
+	sessionUserRollbacksSQL       = "select ss.SID as session_id, se.value as VALUE from v$session ss, v$sesstat se, v$statname sn where se.STATISTIC# = sn.STATISTIC# and se.SID = ss.SID and NAME = 'user rollbacks'"
+	activeSessionTotalSQL         = "select status, type, count(*) as VALUE FROM v$session GROUP BY status, type"
+	cachedSessionTotalSQL         = "select type, count(*) as VALUE FROM v$session WHERE status = 'CACHED' GROUP BY type"
+)
+
+type dbProviderFunc func() (*sql.DB, error)
+
+type clientProviderFunc func(*sql.DB, string, *zap.Logger) dbClient
+
+type scraper struct {
+	logger             *zap.Logger
+	metricsBuilder     *metadata.MetricsBuilder
+	dbProviderFunc     dbProviderFunc
+	clientProviderFunc clientProviderFunc
+	db                 *sql.DB
+	id                 config.ComponentID
+	scrapeCfg          scraperhelper.ScraperControllerSettings
+	startTime          pcommon.Timestamp
+	metricsSettings    metadata.MetricsSettings
+}
+
+func newScraper(id config.ComponentID, metricsBuilder *metadata.MetricsBuilder, metricsSettings metadata.MetricsSettings, scrapeCfg scraperhelper.ScraperControllerSettings, logger *zap.Logger, providerFunc dbProviderFunc, clientProviderFunc clientProviderFunc) *scraper {
+	return &scraper{
+		id:                 id,
+		metricsBuilder:     metricsBuilder,
+		metricsSettings:    metricsSettings,
+		scrapeCfg:          scrapeCfg,
+		logger:             logger,
+		dbProviderFunc:     providerFunc,
+		clientProviderFunc: clientProviderFunc,
+	}
+}
+
+var _ scraperhelper.Scraper = (*scraper)(nil)
+
+func (s *scraper) ID() config.ComponentID {
+	return s.id
+}
+
+func (s *scraper) Start(context.Context, component.Host) error {
+	s.startTime = pcommon.NewTimestampFromTime(time.Now())
+	var err error
+	s.db, err = s.dbProviderFunc()
+	if err != nil {
+		return fmt.Errorf("failed to open db connection: %w", err)
+	}
+	return nil
+}
+
+func (s *scraper) Scrape(ctx context.Context) (pmetric.Metrics, error) {
+	s.logger.Debug("Begin scrape")
+	now := pcommon.NewTimestampFromTime(time.Now())
+	if s.metricsSettings.OracledbQueryCPUTime.Enabled {
+		if err := s.executeOneQueryWithFullText(ctx, now, queryCPUTimeSQL, s.metricsBuilder.RecordOracledbQueryCPUTimeDataPoint); err != nil {
+			return pmetric.Metrics{}, fmt.Errorf("error executing %s: %w", queryCPUTimeSQL, err)
```

Comment:
Is it better to track with a multierr instead of early exiting?

Reply:
Could it be a bit more maintainable ranging over a slice of structs w/ the query, enabled, and recorder instead of each individual item?

Reply:
multierr - certainly. Do we want to collect metrics we can then and still error out? How is that best handled? Do we log the multierr or use the obsreceiver approach?

Reply:
I can refactor this code to make it look less jarring as well.

Reply:
Appears `scrapererror.NewPartialScrapeError()` will cause the correct metric count to be used in the obs report: https://github.com/open-telemetry/opentelemetry-collector/blob/main/receiver/scraperhelper/scrapercontroller.go#L198

Reply:
I think manually logging may be required though to surface to the user

Reply:
OK, implemented the partial scrape error.